### PR TITLE
Switch to upstream windows_exporter

### LINKED ISSUE #1631: Update windows_exporter with latest
Update windows_exporter with latest version of the community windows exporter

### LINKED ISSUE #3364: Update github.com/prometheus-community/windows_exporter to v0.22.0
An update for `github.com/prometheus-community/windows_exporter` (version `v0.22.0`) is now available. Version `v0.16.0` is currently in use.

### LINKED ISSUE #3420: Get changes of `windows_exporter` merged upstream and remove `replace` in go.mod

### PR DESCRIPTION
**Pull request by jkroepke**
#### PR Description

This PR removes the outdated internal grafana/windows_exporter fork.

This PR is still in draft, since various upstream PR needs to be solved first. This PR based on personal forks of windows_exporter which are planned to be merged upstreamed. After that, the PR is ready to review.

I'm creating the PR right now to indicate that some is working on the windows_exporter.

It introduce a breaking change from https://github.com/prometheus-community/windows_exporter/pull/1170 (not sure how to handle this). If Grafana Agent follows SemVer, the breaking change is fine.

Test on a windows 10 system with following config:

```
prometheus.exporter.windows "exporter1" {
  // Subset of overall collectors  
  enabled_collectors = "cpu,cs,logical_disk"
}

prometheus.exporter.windows "exporter2" {
  // Different subset of overall collectors  
  enabled_collectors = "net,os,service,system"
}

prometheus.scrape "example" {
  targets = concat(
    prometheus.exporter.windows.exporter1.targets,
    prometheus.exporter.windows.exporter2.targets,
  )

  forward_to = [...]
}
```

Prometheus UI:

image

#### Which issue(s) this PR fixes

* Fixes #1631 
* Fixes #3364 
* Fixes #3420 
* Relates #3122 (not sure, if the issues exists upstreams)

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by mattdurham** [comment]
Reviewing this and going to run it on my local VM to smoke test it a bit.

File: component/prometheus/exporter/windows/windows_test.go [hunk 1]
```diff
@@ -10,39 +11,40 @@ import (
...
 		exchange {
 			enabled_list = ["example"]
 		}
 		
 		iis {
-			site_whitelist = ".+"
-			site_blacklist = ""
-			app_whitelist = ".+"
-			app_blacklist = ""
+			site_include = ".+"
```
Comment (mattdurham) [inline component/prometheus/exporter/windows/windows_test.go:21]: This is the preferred nomenclature but we should allow the old values and mark them as deprecated.

File: docs/sources/flow/reference/components/prometheus.exporter.windows.md [hunk 2]
```diff
@@ -29,9 +29,10 @@ prometheus.exporter.windows "LABEL" {
 The following arguments can be used to configure the exporter's behavior.
 All arguments are optional. Omitted fields take their default values.
 
-| Name                       | Type           | Description                                                   | Default | Required |
-|----------------------------|----------------|---------------------------------------------------------------|---------|----------|
-| `enabled_collectors`       | string`        | List of collectors to enable.                                |         | no       |
+| Name                 | Type       | Description                               | Default | Required |
+|----------------------|------------|-------------------------------------------|---------|----------|
+| `enabled_collectors` | `string`   | List of collectors to enable.             |         | no       |
+| `timeout`            | `duration` | Configure timeout for collecting metrics. | `4m`    | no       |
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/prometheus.exporter.windows.md:35]: This new timeout gives me a bit of pause, this seems an artifact from windows_exporter running standalone but generally the scrape_timeout will be the driver. Unfortunately there isn't a way to propagate that around. Thoughts on changing the doc to something like `Configure timeout for windows_exporter serving metrics, this does not override any scraping timeouts`, though dang that is wordy. The other path would be to make it hard coded to the default and not expose it for the time being. Any thoughts?

Comment (mattdurham) [reply docs/sources/flow/reference/components/prometheus.exporter.windows.md:35]: Good call, not in love with the verbiage around it but cant think of anything better and we can always add context if we get questions.

Comment (jkroepke) [reply docs/sources/flow/reference/components/prometheus.exporter.windows.md:35]: Okay, then I would reverted timeout change here to the hardcoded one...  adding the context should do in a separate PR?

File: component/prometheus/exporter/windows/windows_test.go (hunk 1 above)
Comment (jkroepke) [reply component/prometheus/exporter/windows/windows_test.go:19]: The upstream windows_exporter leaving deprecation notes. 

https://github.com/prometheus-community/windows_exporter/blob/79781c6d75fc5cef4d28685b17a4b581e038aa10/collector/net.go#L90

I hope thats enough

**Comment by jkroepke** [comment]
@mattdurham I hope I address all you issues!

**Review by mattdurham — APPROVED** [review]
Fantastic work, some small formatting things/deprecation notes but I will fix them post merge.

**Comment by mattdurham** [comment]
@jkroepke can you debug the windows_exporter, I noticed when debugging agent on windows and also windows_exporter directly I get a syscall error. Several others were able to recreate. Running without the debugger is fine.

**Comment by mattdurham** [comment]
And to close this out found the issue https://github.com/leoluk/perflib_exporter/pull/42, if you check the error while running on RegKeyClose it returns invalid handle which for whatever reason crashes when debugger is attached.

**Comment by kago-dk** [comment]
https://github.com/grafana/agent/blob/main/docs/sources/static/configuration/integrations/windows-exporter-config.md should be updated to cover the changes mentioned in the changelog.

**Pull request merged**